### DISCUSSION TITLE
Option entry in second argument of onchange is useless

### DISCUSSION THREAD

**Discussion by dehghani-mehdi** [JedWatson/react-select#5438] [Ideas]
Hi,

I think you should either remove `option` entry form second argument of `onChange` or fill the value in both `single` and `multi` modes.

Right now `option` is `undefined` in `single` mode and filled with selected item when `isMulti` is `true`. and this makes `option` useless, _unless there is usecase which I didn't catch_

Single mode:
`{ action: "select-option", option: undefined, name: "blah" }`

Multiple mode:
`{ action: "select-option", option: {…}, name: "blah" }`

**Comment by WaqqarSuleman** [comment]
It would be consistent if the `option` entry from the second argument is populated in case of both single and multi select.

**Comment by ebonow** [comment]
Greetings @WaqqarSuleman ,

I recently brought this up and agree. We recently accepted a PR which added `option` into the acitonMeta for `create-option` and think it would be much more consistent if the option was always present for `select-option` regardless of it being single or multi.

**Comment by WaqqarSuleman** [comment]
> Greetings @WaqqarSuleman ,
> 
> I recently brought this up and agree. We recently accepted a PR which added `option` into the acitonMeta for `create-option` and think it would be much more consistent if the option was always present for `select-option` regardless of it being single or multi.

@ebonow You are totally correct!

**Comment by ksoldau** [comment]
Has there been any movement to add `option` to `select-option` for the single case?
